src/meshes/tree_mesh.jl

```diff
@@ -119,9 +124,19 @@ function TreeMesh(coordinates_min::NTuple{NDIMS, Real},
         throw(ArgumentError("`initial_refinement_level` must be a non-negative integer (provided `initial_refinement_level = $initial_refinement_level`)"))
     end
 
+    # Check if elements in coordinates_min and coordinates_max are all of type RealT
+    for i in 1:NDIMS
+        if !(coordinates_min[i] isa RealT)
+            @warn "Element $i in `coordinates_min` is not of type $RealT (provided `coordinates_min[$i] = $(coordinates_min[i])`)"
+        end
+        if !(coordinates_max[i] isa RealT)
+            @warn "Element $i in `coordinates_max` is not of type $RealT (provided `coordinates_max[$i] = $(coordinates_max[i])`)"
+        end
+    end
+
```

Comment:
This is a breaking change - so many users will stick to previous usage like
```julia
coordinates_min = (-2.0f0, -2.0f0)
coordinates_max = (2.0f0, 2.0f0)
mesh = TreeMesh(coordinates_min, coordinates_max,
                initial_refinement_level = 4,
                n_cells_max = 10_000)
```
expecting a `Float32` output but to get a warning that an element is not of type `Float64` (where does `Float64` come from?). It would be better if the warning were more like
```julia
Input types do not match the default `Float64` or your specified type. Use the keyword `RealT = type` to set the type.
```

Reply:
Is it? I mean we only throw a warning, the rest works just the same.

Reply:
It is just a suggestion - both are ok for me